### PR TITLE
chore(tests): Fix TS errors and re-enable `ts-jest`'s `diagnostics` option

### PR DESCRIPTION
**Pull request by trevor-scheer**
Fixes #661

### CODE REVIEW

File: gateway-js/src/index.ts [hunk 1]
```diff
@@ -800,7 +800,7 @@ export class ApolloGateway implements GraphQLService {
       this.state = {
         phase: 'waiting to poll',
         doneWaiting,
-        pollWaitTimer: setTimeout(() => {
+        pollWaitTimer: global.setTimeout(() => {
```
Comment (trevor-scheer) [inline gateway-js/src/index.ts:803]: @glasser Is this ok? `ts-jest` can't differentiate between `setTimeout` in the browser vs. in node, and they have different return types.

Comment (glasser) [reply gateway-js/src/index.ts:803]: Hmm. This is actually a change to non-test code? Maybe we need somebody who is more of a TS/Jest expert to weigh in.

Comment (glasser) [reply gateway-js/src/index.ts:803]: I think the trick is to remove `dom` from the `lib` in tsconfig.test.base.json.

However, that makes an error when trying to build `node_modules/apollo-link-http-common`, as used by `apollo-link-http` as used in `reporting.test.ts`.

That's an ancient version of Apollo Link from AC1 or AC2 or something; you could try upgrading to use `HttpLink` from `@apollo/client` instead. Or frankly just change the single use of this link to use `fetch` like `endToEnd.test.ts` does?

**Review by glasser — CHANGES_REQUESTED** [review]
i forgot to hit this button before :)

**Comment by trevor-scheer** [comment]
Closing in favor of #1376

**Pull request closed without merging**